src/Navigation/INavigationEngine.ts

```diff
@@ -156,12 +156,42 @@ export interface INavigationEnginePlugin {
      */
     getMaximumSubStepCount(): number;
 
+    /**
+     * Creates a cylinder obstacle and add it to the navigation
+     * @param position world position
+     * @param radius cylinder radius
+     * @param height cylinder height
+     * @returns the obstacle freshly created
+     */
+    addCylinderObstacle(position: Vector3, radius: number, height: number): IObstacle;

```

Comment:
Does this signature mean the cylinder is always parallel to the Y axis? We can't have an arbitrary rotated cylinder?

Reply:
No ellipse. But, a PI/2 on X or Z axis gives a rectangle, so you can use a box for that case ;)